ui/utils/utils.js

```diff
@@ -387,3 +388,16 @@ export const processDesign = (design) => {
     designJson,
   };
 };
+
+export const getDesignVersion = (design) => {
+  if (design.visibility === 'published') {
+    return design.catalog_data.published_version;
+  } else {
+    try {
+      const parsedYaml = yaml.load(design.pattern_file);
+      return parsedYaml.version;
+    } catch (error) {
+      console.error('Error parsing YAML:', error);
```

Comment:
Hey can you return "0.0.1" as default design version in case there is issue in reading version
```suggestion
      console.error('Error parsing YAML:', error);
      return DEFAULT_DESIGN_VERSION
```

Create a new const var in constant file in utils with name DEFAULT_DESIGN_VERSION and set value to "0.0.1"

Reply:
Thanks for making sure we have this in utils

Reply:
We don't put a `v` here, right?

Reply:
I'll argue that the UI should never attempt to define this. The design either has a version or it doesn't. If it doesn't, the first time / next time that it is opened in Meshery Server, it will.

Reply:
Don't through an error saying that you can't parse the YAML. You can. It's just missing a version.

Reply:
Handle it gracefully and move on. Thanks for handling is not giving it a version number that it doesn't actually have. Graceful handling is convey to the user that it is absent a version number (e.g. `N/A`) and not erroring out.

Reply:
I wonder if we want to show this error in toaster to user that's why we are putting this in console as 3rd level error, we are already making sure that if version is not present we are not showing it in info modal

Reply:
However we can improve the error, @Kevin222004 I'm resolving this conversation considering to not return version, thankyou Lee